### PR TITLE
HYDRATOR-703 better error message

### PR DESCRIPTION
**Pull request by gsps1**
add cause of exception during plugin method call in the http response.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java [hunk 1]
```diff
@@ -474,7 +474,11 @@ public void callArtifactPluginMethod(HttpRequest request, HttpResponder responde
       } else if (e.getCause() instanceof javax.ws.rs.BadRequestException) {
         throw new BadRequestException(e.getCause());
       } else {
-        responder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR, "Error while invoking plugin method");
+        String message = "Error while invoking plugin method.";
+        if (e.getCause() != null && e.getCause().getMessage() != null) {
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java]: maybe get the root cause?

**Comment by gsps1** [comment]
@albertshau updated to get root case.

**Review by albertshau — APPROVED** [review]
lgtm

**Comment by gsps1** [comment]
http://builds.cask.co/browse/CDAP-DUT5577 - green - merging

**Pull request merged**